plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java

```diff
@@ -115,15 +117,16 @@ void setSiteUri(URI siteUri) {
   }
 
   public synchronized List<LibraryFile> getLibraryFiles() {
-    return new ArrayList<>(libraryFiles);
+    return new ArrayList<>(transitiveDependencies);
   }
 
   /**
    * @param libraryFiles artifacts associated with this library, cannot be <code>null</code>
    */
   public synchronized void setLibraryFiles(List<LibraryFile> libraryFiles) {
     Preconditions.checkNotNull(libraryFiles);
-    this.libraryFiles = new ArrayList<>(libraryFiles);
+    this.directDependencies = new ArrayList<>(libraryFiles);
+    this.transitiveDependencies = this.directDependencies;
```

Comment:
This needs some thoughts, as it blindly overwrites the transitiveDependencies. It's getting hard to track the effect, and it's not clear what the contract of this method should be. However, I think we should set `resolved = false` here. I wish everything were immutable.

Reply:
Immutability isn't really feasible for this class because we need to load the transitive dependencies when used. Otherwise we waste a lot of time and network bandwidth loading every dependency of every library even though the user doesn't touch it.

resolved should not be set to false here because this is only called as part of initialization of a new library object. 

I might be able to not set transitiveDependencies here and instead initialize it lazily when getLibraryFiles is called. I could even call resolveDependencies from inside getLibraryFiles though that will have the side effect of making a few more of our tests ht the network. Still that hasn't been a big problem so far, so maybe that makes sense.

Reply:
At least the code is same (I see it is just renaming) and there is no behavioral change, so I think this won't break things, but looks like we should have set `resolved = false` to be correct.

Reply:
Never mind. I think I misunderstood how things work here.